tests/Integration/Events/ShouldDispatchAfterCommitEventTest.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+namespace Illuminate\Tests\Integration\Events;
+
+use Illuminate\Contracts\Events\ShouldDispatchAfterCommit;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Event;
+use Mockery as m;
+use Orchestra\Testbench\TestCase;
+
+class ShouldDispatchAfterCommitEventTest extends TestCase
```

Comment:
One test case that comes to my mind:

* Dispatch all events, including ones registered within nested transactions, **only after** the root transaction commits.

Is this already handled somehow by `$afterCommit` implementation? It is important to do a sanity check, here.

Reply:
That's a good catch. I'm adding some tests.
If you check `DatabaseTransactionsManager`. It only executes the callbacks once the all the transactions have been handled. This is the `ManagesTransactions::commit()` code:

```php
    public function commit()
    {
        if ($this->transactionLevel() == 1) {
            $this->fireConnectionEvent('committing');
            $this->getPdo()->commit();
        }

        $this->transactions = max(0, $this->transactions - 1);

        if ($this->afterCommitCallbacksShouldBeExecuted()) {
            $this->transactionsManager?->commit($this->getName());
        }

        $this->fireConnectionEvent('committed');
    }
```

Reply:
@fntneves Added the test [here](https://github.com/laravel/framework/pull/48705/commits/984f9aac923488b55cfb045d939e3b07d1ebc618), let me know if that was in line with what you were thinking!

Reply:
That is! Thank you, you're nailing it!